tsdb/chunks/head_chunks.go

```diff
@@ -696,7 +720,11 @@ func (cdm *ChunkDiskMapper) Chunk(ref ChunkDiskMapperRef) (chunkenc.Chunk, error
 
 	// Encoding.
 	chkEnc := mmapFile.byteSlice.Range(chkStart, chkStart+ChunkEncodingSize)[0]
-
+	sourceChkEnc := chunkenc.Encoding(chkEnc)
+	if cdm.IsOutOfOrderChunk(sourceChkEnc) {
```

Comment:
You don't need this if check. You can simply just apply the bit mask everytime.

Reply:
understood